app/scheduled/circulars/uploadTar.ts

```diff
@@ -0,0 +1,119 @@
+/*!
+ * Copyright © 2023 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3'
+import { createReadableStreamFromReadable } from '@remix-run/node'
+import { Readable } from 'stream'
+import type { Pack } from 'tar-stream'
+import { pack as tarPack } from 'tar-stream'
+
+import type { CircularActionContext } from './circularAction'
+import { getEnvOrDie } from '~/lib/env.server'
+import type { Circular } from '~/routes/circulars/circulars.lib'
+import { formatCircular } from '~/routes/circulars/circulars.lib'
+
+interface TarContextObject {
+  pack: Pack
+  readableTar: Readable | ReadableStream<Uint8Array>
+  fileType: string
+}
+
+async function uploadStream(tarContext: TarContextObject) {
+  const s3 = new S3Client({})
+  const Bucket = getEnvOrDie('ARC_STATIC_BUCKET')
+
+  await s3.send(
+    new PutObjectCommand({
+      Bucket,
+      Key: `circulars.archive.${tarContext.fileType}.tar`,
+      Body: tarContext.readableTar,
+    })
+  )
+}
+
+export async function makeTarFile({
+  pack,
+  readableTar,
+  circularArray,
+  fileType,
+}: {
+  pack: any
+  readableTar: Readable
+  circularArray: Circular[]
+  fileType: 'txt' | 'json'
+}) {
+  for (const circular of circularArray) {
+    if (fileType === 'txt') {
+      const txt_entry = pack.entry(
+        { name: `archive.txt/${circular.circularId}.txt` },
+        formatCircular(circular)
+      )
+      txt_entry.end()
+    } else if (fileType === 'json') {
+      delete circular.sub
+      const json_entry = pack.entry(
+        { name: `archive.json/${circular.circularId}.json` },
+        JSON.stringify(circular, null, 2)
+      )
+      json_entry.end()
+    }
+  }
+  return { context: { pack, readableTar } }
+}
```

Comment:
There are some refactorings in #1288 that make this a lot simpler. See https://github.com/nasa-gcn/gcn.nasa.gov/blob/c3b53c2086d550457813a3acfd5a0ad772db2e0b/app/routes/circulars/circulars.server.ts#L273-L318.